go/vt/mysqlctl/backup.go

```diff
@@ -291,6 +299,43 @@ func ShouldRestore(ctx context.Context, params RestoreParams) (bool, error) {
 	return checkNoDB(ctx, params.Mysqld, params.DbName)
 }
 
+// ensureRestoredGTIDPurgedMatchesManifest sees o the following: when you restore a full backup, you want the MySQL server to have
+// @@gtid_purged == <gtid-of-backup>. This then also implies that @@gtid_executed equals same value. This is because we restore without
+// any binary logs.
+func ensureRestoredGTIDPurgedMatchesManifest(ctx context.Context, manifest *BackupManifest, params *RestoreParams) error {
+	if manifest == nil {
+		return nil
+	}
+	if manifest.Position.GTIDSet == nil {
+		return nil
+	}
+	gtid := manifest.Position.GTIDSet.String()
+	if gtid == "" {
+		return nil
+	}
+	// Xtrabackup 2.4's restore seems to set @@gtid_purged to be the @@gtid_purged at the time of backup. But this is not
+	// the desired value. We want to set @@gtid_purged to be the @@gtid_executed of the backup.
+	// As reminder, when restoring from a full backup, setting @@gtid_purged also sets @@gtid_executed.
+	restoredGTIDPurgedPos, err := params.Mysqld.GetGTIDPurged(ctx)
+	if err != nil {
+		return vterrors.Wrapf(err, "Reading gtid_purged after restore")
```

Comment:
Nit, but error messages should not be capitalized.
```suggestion
		return vterrors.Wrapf(err, "failed to read gtid_purged after restore")
```